### PR TITLE
Add start and end vertex label ids and names in edge object

### PR DESCRIPTION
**Pull request by Zainab-Saad**
A sample of modified edge object is 
```
{"id": 1, "label": "PARTICIPATES", "end_id": 2, "start_id": 1, "properties": {"from": 2021}, "end_label_id": 9, "end_label_name": "Project", "start_label_id": 3, "start_label_name": "Developer"}
```
14 of 26 regression tests fail after these changes (none have been fixed in this PR yet)

### CODE REVIEW

**Comment by rafsun42** [comment]
@Zainab-Saad Thank you.

Could you update the test files that were affected by this PR please?

**Comment by Zainab-Saad** [comment]
@rafsun42 
I have fixed the failing regression tests and done some code changes mainly for:
- making MATCH work without using `_extract_label_id` (removed it)
- Updated `age_vle` and edge/path `typecasting` to reflect the changes in edge object

Now 3 of 26 tests fail (`expr, age_load, graph_generation`) which will be fixed by PR # 26 and 29

**Comment by rafsun42** [comment]
@Zainab-Saad 

Looks good. However- 

1. `graph_generation` has errors `ERROR:  graph_oid and label_id must not be null`.
2. `age_load` also has errors. But I assume the errors will be fixed in age_load's PR. Is that correct?

File: regress/expected/agtype.out [hunk 1]
```diff
@@ -1629,32 +1629,14 @@ NOTICE:  graph "orderability_graph" has been created
...
+SELECT * FROM cypher('orderability_graph', $$ CREATE (:vertex {prop: null}), (:vertex {prop: 1}), (:vertex {prop: 1.01}),(:vertex {prop: true}), (:vertex {prop:"string"}),(:vertex {prop:"string_2"}), (:vertex {prop:[1, 2, 3]}), (:vertex {prop:[1, 2, 3, 4, 5]}), (:vertex {prop:{bool:true, i:0}}), (:vertex {prop:{bool:true, i:null}}), (:vertex {prop: {id:0, label: "v", properties:{i:0}}::vertex}),  (:vertex {prop: {id: 2, start_id: 0, end_id: 1, label: "e", properties: {i: 0}, end_label_id: 3, end_label_name: "vertex", start_label_id: 3, start_label_name: "vertex"}::edge}), (:vertex {prop: [{id: 0, label: "v", properties: {i: 0}}::vertex, {id: 2, start_id: 0, end_id: 1, label: "e", properties: {i: 0}, end_label_id: 3, end_label_name: "vertex", start_label_id: 3, start_label_name: "vertex"}::edge, {id: 1, label: "v", properties: {i: 0}}::vertex]::path}) $$)  AS (x agtype);
  x 
 ---
 (0 rows)
 
-SELECT * FROM cypher('orderability_graph', $$ MATCH (n) RETURN n ORDER BY n.prop $$) AS (sorted agtype);
```
Comment (rafsun42) [inline regress/expected/agtype.out:1637]: Is this change expected?

Comment (rafsun42) [inline regress/expected/agtype.out:1637]: Why is this test removed?

Comment (Zainab-Saad) [reply regress/expected/agtype.out:1637]: it's a mistake, I have fixed that

**Comment by Zainab-Saad** [comment]
> 1. graph_generation has errors ERROR:  graph_oid and label_id must not be null. 2. age_load also has errors. But I assume the errors will be fixed in age_load's PR. Is that correct?

Yes, these errors will be fixed with PR on age_load

File: regress/expected/agtype.out (hunk 1 above)
Comment (rafsun42) [reply regress/expected/agtype.out:1637]: @Zainab-Saad Okay. Are you sure that rest of the tests are correct?

**Pull request merged**